packages/proxyIdentity/contracts/OfferableIdentity.sol

```diff
@@ -0,0 +1,89 @@
+pragma solidity 0.8.3;
+
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "./IdentityManager.sol";
+import "./IOfferable.sol";
+
+
+contract OfferableIdentity is IOfferable, ERC165 {
+  address public owner;
+  address manager;
+  address ethrRegistry;
+
+  address public offeredTo;
+
+  event TransactionSent(bytes indexed data, uint256 indexed value);
+
+  function init(address _owner, address _ethrRegistry) external {
+    require(
+      manager == address(0),
+      "OfferableIdentity: Identity can be initialize only once"
+    );
+    owner = _owner;
+    manager = msg.sender;
+    ethrRegistry = _ethrRegistry;
+    IdentityManager(manager).identityCreated(owner);
+  }
+
+  modifier isOwner() {
+    require(msg.sender == owner, "OfferableIdentity: Only owner allowed");
+    _;
+  }
+
+  modifier isOfferedTo() {
+    require(offeredTo != address(0), "OfferableIdentity: Proxy is not offered");
+    require(
+      msg.sender == offeredTo,
+      "OfferableIdentity: Proxy offered to other account"
+    );
+    _;
+  }
+
+  function offer(address _offeredTo) external override isOwner {
+    offeredTo = _offeredTo;
+    IdentityManager(manager).identityOffered(offeredTo);
+  }
+
+  function acceptOffer() external override isOfferedTo {
+    owner = offeredTo;
+    IdentityManager(manager).identityAccepted(offeredTo);
+    closeOffer();
+  }
+
+  function rejectOffer() external override isOfferedTo {
+    IdentityManager(manager).identityRejected(offeredTo);
+    closeOffer();
+  }
+
+  function cancelOffer() external override isOwner {
+    IdentityManager(manager).identityOfferCanceled(offeredTo);
+    closeOffer();
+  }
+
+  function closeOffer() internal {
+    offeredTo = address(0);
+  }
+
+  function update(bytes memory data, uint256 value)
+    external
+    isOwner
+    override
+    returns (bool success)
+  {
+    (success, ) = ethrRegistry.call{ gas: gasleft() - 5000, value: value}(data);
+    require(success, "OfferableIdentity: Document is not updated");
+    emit TransactionSent(data, value);
+  }
```

Comment:
@JGiter After speaking with Micha, he confirmed that this should be a generic proxy function (as you had it before). Could you please revert it to the generic function? This is for a couple of reasons:

- The asset should be able to accommodate future use cases (such as being able to notify of energy production)
- If the `ethrRegistry` needs to be redeployed then we will not need to redeploy all of the asset contracts 

@ngyam Does this seem reasonable to you?

Reply:
Reverted. Also added setter to avoid redeploying when registry is changed

Reply:
Reverted. To allow call arbitrary contract I just added its address to params

Reply:
If you have a good reason for this, sure. It looks okay like this.